### PR TITLE
Container Development with VS2019, Git and Docker Desktop

### PR DESCRIPTION
**Pull request by mahendra-shinde**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

File: 101-vs2019-git-docker-windows2019/Readme.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+Build containerized applications using both docker-desktop and visual studio 2019 latest community edition. The 'docker-desktop' installation using custom script extension would take around 15 to 20 minutes.
+
+## Applications Installed
+
+- Visual Studio CODE 
+- Visual Studio 2019 Latest Community Edition
+- Git for Windows
+- Docker Desktop 
+
+Please restart the Virtual machine once deployment is completed. Once restarted, launch docker-deskop from start menu. First run of docker-desktop would deploy Virtual machine for linux containers. It might add up another 5-10 minutes.
```
Comment (bmoore-msft) [inline 101-vs2019-git-docker-windows2019/Readme.md:28]: if this is required, why not restart from the script ?

File: 101-vs2019-git-docker-windows2019/Readme.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+[![Deploy To Azure US Gov](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazuregov.svg?sanitize=true)](https://portal.azure.us/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F101-vs2019-git-docker-windows2019%2Fazuredeploy.json)
+[![Visualize](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/visualizebutton.svg?sanitize=true)](http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F101-vs2019-git-docker-windows2019%2Fazuredeploy.json)   
+
+
+
+Build containerized applications using both docker-desktop and visual studio 2019 latest community edition. The 'docker-desktop' installation using custom script extension would take around 15 to 20 minutes.
```
Comment (bmoore-msft) [inline 101-vs2019-git-docker-windows2019/Readme.md:19]: Capitalize the product name (i.e. Visual Studio)

"would take" or "will take"?

File: 101-vs2019-git-docker-windows2019/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,262 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "storagename": {
+            "type": "string",
```
Comment (bmoore-msft) [inline 101-vs2019-git-docker-windows2019/azuredeploy.json:6]: add a defaultValue using uniqueString

File: 101-vs2019-git-docker-windows2019/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+            "metadata": {
+                "description":"Unique Storage Account name"
+            }
+        },
+        "vm-dns": {
+            "type": "string",
```
Comment (bmoore-msft) [inline 101-vs2019-git-docker-windows2019/azuredeploy.json:12]: add a defaultValue using uniqueString

File: 101-vs2019-git-docker-windows2019/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+            "metadata": {
+                "description": "VM DNS Label prefix"
+            }
+        },
+        "adminUser": {
+            "type": "string",
+            "defaultValue":"mahendra",
```
Comment (bmoore-msft) [inline 101-vs2019-git-docker-windows2019/azuredeploy.json:19]: remove defaultValue

File: 101-vs2019-git-docker-windows2019/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
+                "description": "SAS Token for accessing script path"
+            }
+        }
+    },
+    "variables":{
+        "scriptUrl": "[uri(parameters('_artifactsLocation'),concat('installscript.ps1',parameters('_artifactsLocationSasToken')))]"
+    },
+    "resources": [    
+        {
+            "name": "[toLower(parameters('storagename'))]",
```
Comment (bmoore-msft) [inline 101-vs2019-git-docker-windows2019/azuredeploy.json:69]: See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#sort-order-of-properties for the proper sort order of properties

File: 101-vs2019-git-docker-windows2019/metadata.json [hunk 7]
```diff
@@ -0,0 +1,10 @@
+{
+    "$schema": "https://aka.ms/azure-quickstart-templates-metadata-schema#",
+    "type": "QuickStart",
+    "itemDisplayName": "Visual Studio 2019 CE with Docker Desktop",
+    "description": " Container Development for Visual Studio 2019 CE with Docker Desktop",
```
Comment (bmoore-msft) [inline 101-vs2019-git-docker-windows2019/metadata.json:5]: "with Visual Studio..."

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: 101-vs2019-git-docker-windows2019/Readme.md (hunk 1 above)
Comment (mahendra-shinde) [reply 101-vs2019-git-docker-windows2019/Readme.md:28]: It's mentioned in azure docs, not to restart VM from custom script.
https://docs.microsoft.com/en-us/azure/virtual-machines/extensions/custom-script-windows

File: 101-vs2019-git-docker-windows2019/metadata.json (hunk 7 above)
Comment (mahendra-shinde) [reply 101-vs2019-git-docker-windows2019/metadata.json:5]: Fixed, thanks

File: 101-vs2019-git-docker-windows2019/azuredeploy.json (hunk 6 above)
Comment (mahendra-shinde) [reply 101-vs2019-git-docker-windows2019/azuredeploy.json:69]: Done, for all resources in template.

File: 101-vs2019-git-docker-windows2019/Readme.md (hunk 2 above)
Comment (mahendra-shinde) [reply 101-vs2019-git-docker-windows2019/Readme.md:19]: Fixed, Thanks

File: 101-vs2019-git-docker-windows2019/azuredeploy.json (hunk 3 above)
Comment (mahendra-shinde) [reply 101-vs2019-git-docker-windows2019/azuredeploy.json:6]: Fixed.

File: 101-vs2019-git-docker-windows2019/azuredeploy.json (hunk 4 above)
Comment (mahendra-shinde) [reply 101-vs2019-git-docker-windows2019/azuredeploy.json:12]: Fixed

File: 101-vs2019-git-docker-windows2019/azuredeploy.json (hunk 5 above)
Comment (mahendra-shinde) [reply 101-vs2019-git-docker-windows2019/azuredeploy.json:19]: Fixed

File: 101-vs2019-git-docker-windows2019/Readme.md (hunk 1 above)
Comment (mahendra-shinde) [reply 101-vs2019-git-docker-windows2019/Readme.md]: I have updated my script to start docker on every login using ScheduledTask.

**Pull request merged**